### ISSUE TITLE
Not able to go full-screen in video players, like YouTube etc

### ISSUE THREAD

**Issue by ishaanbhimwal** [black7375/Firefox-UI-Fix#983]
### Describe the bug

Full-screen mode is broken due to lepton but works fine if customization settings are set to false. I was using the default `user.js`.

### OS

Linux

### Firefox Version

129.0.1

### Distribution

- [X] [Original Lepton](https://github.com/black7375/Firefox-UI-Fix)
- [ ] [Lepton's photon style](https://github.com/black7375/Firefox-UI-Fix/tree/photon-style)
- [ ] [Lepton's proton style](https://github.com/black7375/Firefox-UI-Fix/tree/proton-style)

### Theme

- [ ] Light
- [ ] Dark
- [ ] Alpenglow (System: Light)
- [ ] Alpenglow (System: Dark)
- [X] System Default
- [ ] Others: (e.g. [Photon Colors](https://addons.mozilla.org/en-US/firefox/addon/photon-colors/), [Google Chrome Dark](https://addons.mozilla.org/en-US/firefox/addon/google-chrome-dark/), [Fairytale Of Nature](https://addons.mozilla.org/en-US/firefox/addon/fairytale-of-nature/))

### `user.js` setup

<details><summary><strong><code>user.js</code> setup</strong></summary><br>

```javascript
// Your settings
```
</details>

**Comment by black7375** [comment]
YouTube full screen works well for me. What symptoms did you have?

**Comment by ishaanbhimwal** [comment]
Can I use [Betterfox js](https://github.com/yokoffing/Betterfox) alongside Lepton ui?

**Comment by black7375** [comment]
Of course it is possible.
Waterfox is an example of using these two together.
- https://www.waterfox.net/

**Comment by ShinyGil** [comment]
Do you use fractional scaling? For me, it is Lepton UI when using fractional scaling (e.g. 125%) that causes this.

**Comment by ishaanbhimwal** [comment]
@ShinyGil Yup! I use fractional scaling at 125% as well. For now, I am just using betterfox user.js with the default ui.

**Comment by ShinyGil** [comment]
> @ShinyGil Yup! I use fractional scaling at 125% as well. For now, I am just using betterfox user.js with the default ui.

Well then. @black7375 I believe we found what causes this bug and how to reproduce it.

**Comment by black7375** [comment]
Hello, How can I reproduce this issue?
It seems to work normally with fractional scaling.
Are there any other conditions?

https://github.com/user-attachments/assets/38b0240c-94a5-4d32-822d-ddb9e2bdc7f7

**Comment by ShinyGil** [comment]
> Hello, How can I reproduce this issue? It seems to work normally with fractional scaling. Are there any other conditions?
> schedular-2024-11-28_22.38.18.mp4

Interesting...what distribution is this and are you using Wayland with Firefox in Wayland mode? This happened to me on both Arch and still happens on Fedora, so I wouldn't expect that to be relevant, but you never know.

**Comment by ShinyGil** [comment]
> I am using EndeavourOS(KDE, x-window). Can this be replicated in a Wayland environment? Could you please tell me what kind of environment it is?

I am using KDE Wayland. I can replicate this 100% of the time in Wayland with Firefox in Wayland mode.

Some builds of Firefox launch in XWayland when on Wayland (I don't remember if this is the case for Arch's), but the environment variable `MOZ_ENABLE_WAYLAND=1` will force it to run in Wayland.
